File: app/src/main/java/br/com/jonathanzanella/myexpenses/transaction/Transaction.java
package br.com.jonathanzanella.myexpenses.transaction;

import org.joda.time.DateTime;

import java.text.SimpleDateFormat;
import java.util.Locale;

public interface Transaction {
	SimpleDateFormat sdf = new SimpleDateFormat("dd/MM/yy", Locale.getDefault());

	String getName();
	DateTime getDate();
	int getAmount();
	boolean credited();
	boolean debited();
}

File: app/src/main/kotlin/br/com/jonathanzanella/myexpenses/transaction/Transaction.kt
package br.com.jonathanzanella.myexpenses.transaction

import org.joda.time.DateTime
import java.text.SimpleDateFormat
import java.util.*

interface Transaction {
    var uuid: String?
    val name: String?
    val amount: Int
    fun credited(): Boolean
    fun debited(): Boolean
    fun getDate(): DateTime

    companion object {
        val SIMPLE_DATE_FORMAT = SimpleDateFormat("dd/MM/yy", Locale.getDefault())
    }
}
